denis.kharisov/task-2-1/cmd/service/main.go

```diff
@@ -0,0 +1,101 @@
+package main
+
+import (
+	"errors"
+	"fmt"
+)
+
+type operation string
+
+const (
+	lessOrEqualOperation    operation = "<="
+	greaterOrEqualOperation operation = ">="
+)
+
+const (
+	defaultMaxTemperature = 30
+	defaultMinTemperature = 15
+)
+
+var (
+	errInvalidRange     = errors.New("max temperature < min temperature")
+	errUnknownOperation = errors.New("unknown operation")
+)
+
+type temperatureRange struct {
+	maxTemperature int
+	minTemperature int
+}
+
+func NewTemperatureRange() *temperatureRange {
+	return &temperatureRange{
+		maxTemperature: defaultMaxTemperature,
+		minTemperature: defaultMinTemperature,
+	}
+}
+
+func (tempRange *temperatureRange) optimalTemperature(operType operation, temperature int) (int, error) {
+	switch operType {
+	case greaterOrEqualOperation:
+		if temperature >= tempRange.minTemperature {
+			tempRange.minTemperature = temperature
+		}
+
+	case lessOrEqualOperation:
+		if temperature <= tempRange.maxTemperature {
+			tempRange.maxTemperature = temperature
+		}
+
+	default:
+		return -1, errUnknownOperation
```

Comment:
```suggestion
		return 0, errUnknownOperation
```

лучше отдавать значение по умолчанию, потому что в данном случае `-1` - это сигнальное значение для конкретного кейса